app/models/listing.rb

```diff
@@ -4,8 +4,11 @@ class Listing < ApplicationRecord
   translates :title
   translates :description, type: :text
 
+  belongs_to :person
+  belongs_to :person_as_provider, class_name: "Person", foreign_key: "person_id"
+  belongs_to :person_as_receiver, class_name: "Person", foreign_key: "person_id"
```

Comment:
Yeah, this was an area I was hoping we'd get in to bc I started by just parent class connections, but for the Profile we already need to separate them and have `person.offers` and `person.asks.`

But imo, we would want to use `person.listings` to populate some sort of engagement stat on the Profile. Yeah, we could get that by `person.asks.count` + `person.listings.count.`

I'm willing to go w this for now. It did feel a little funny.

Reply:
I was trying to avoid namespacing conflict w `Matches,` but maybe this isn't an issue. Testing more now.

Reply:
But don't we want to be able to take any `Listing` and call `.person` on it, rather than have to go find the `type` to find the correct assn method to call?

Reply:
In thinking about this more, my vote would be to not use the `provider` and `receiver` words until there's a `Match?` Until that point, they are a people with asks and/or offers.